### PR TITLE
admin tools initial release

### PR DESCRIPTION
**Pull request by MilosKozak**

### CODE REVIEW

File: static/admin/js/admin.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+(function () {
+  'use strict';
+  //for the tests window isn't the global object
+  var $ = window.$;
+  var _ = window._;
```
Comment (jasoncalabrese) [inline static/admin/js/admin.js]: Codacy Issue found: ['_' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033828/issues/source?bid=2452379&fileBranchId=2865970#l5)

File: lib/admin_plugins/index.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  plugins.doAction = function doAction(event) {
+    var plugin = $(this).attr('plugin');
+    var action = $(this).attr('action');
+    var a = plugins(plugin).actions[action];
+    var ok = true;
+    if (a.confirmText) {
+      ok = window.confirm(Nightscout.client.translate(a.confirmText));
+    }
+    if (ok) {
+      console.log('Running action', action, 'on plugin', plugin)
```
Comment (jasoncalabrese) [inline lib/admin_plugins/index.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033757/issues/source?bid=2452379&fileBranchId=2865970#l62)

File: lib/admin_plugins/cleanstatusdb.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    if (callback) {
+      callback();
+    }
+  }).fail(function fail() {
+    $status.hide().text(translate('Error')).fadeIn('slow');
+    if (callback) {
+      callback();
+    }
+  });
+}
```
Comment (jasoncalabrese) [inline lib/admin_plugins/cleanstatusdb.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033756/issues/source?bid=2452379&fileBranchId=2865970#l69)

File: static/admin/js/admin.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  var moment = window.moment;
+  var Nightscout = window.Nightscout;
+  var client = Nightscout.client;
+  var translate = client.translate;
```
Comment (jasoncalabrese) [inline static/admin/js/admin.js]: Codacy Issue found: ['translate' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033828/issues/source?bid=2452379&fileBranchId=2865970#l9)

File: lib/admin_plugins/index.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        $('<style>')
+          .prop('type', 'text/css')
+          .html(p.css)
+          .appendTo('head');
+      }
+    });
+    $('.adminButton').click(plugins.doAction);
+  };
+
+  plugins.doAction = function doAction(event) {
```
Comment (jasoncalabrese) [inline lib/admin_plugins/index.js]: Codacy Issue found: ['event' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033757/issues/source?bid=2452379&fileBranchId=2865970#l53)

File: static/admin/js/admin.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+(function () {
+  'use strict';
+  //for the tests window isn't the global object
+  var $ = window.$;
+  var _ = window._;
+  var moment = window.moment;
```
Comment (jasoncalabrese) [inline static/admin/js/admin.js]: Codacy Issue found: ['moment' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033828/issues/source?bid=2452379&fileBranchId=2865970#l6)

File: static/admin/js/admin.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+(function () {
+  'use strict';
+  //for the tests window isn't the global object
+  var $ = window.$;
```
Comment (jasoncalabrese) [inline static/admin/js/admin.js]: Codacy Issue found: ['$' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982033828/issues/source?bid=2452379&fileBranchId=2865970#l4)

File: lib/admin_plugins/cleanstatusdb.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
```
Comment (jasoncalabrese) [inline lib/admin_plugins/cleanstatusdb.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/982324139/issues/source?bid=2452379&fileBranchId=2865970#l69)

File: lib/admin_plugins/cleanstatusdb.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      , confirmText: 'Delete all documents from devicestatus collection?'
+    }
+  ];
+
+cleanstatusdb.actions[0].init = function init(client, callback) {
+  var translate = client.translate;
+  var $status = $('#admin_' + cleanstatusdb.name + '_0_status');
+  
+  $status.hide().text(translate('Loading database ...')).fadeIn('slow');
+  $.ajax('/api/v1/devicestatus.json?count=500000', {
```
Comment (jasoncalabrese) [inline lib/admin_plugins/cleanstatusdb.js]: an expensive way to get the count, but don't know if we would want to add an api for it, maybe lower the count to 500 or something, and have something like  "Database contains > 500 records"

Comment (MilosKozak) [inline lib/admin_plugins/cleanstatusdb.js]: yeah. this plugin is an example not written much efficient way. I checked it on my database (never cleaned) and it was something like 5MB. no so bad

Comment (MilosKozak) [inline lib/admin_plugins/cleanstatusdb.js]: but voting for some general solution of getting count

File: lib/admin_plugins/futuretreatments.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+      , description: 'This task find and remove treatments in the future.'
+      , buttonLabel: 'Remove treatmens in the future'
+    }
+  ];
+
+futuretreatments.actions[0].init = function init(client, callback) {
+  var translate = client.translate;
+  var $status = $('#admin_' + futuretreatments.name + '_0_status');
+  
+  function showTreatments(treatments, table) {
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futuretreatments.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/983074590/issues/source?bid=2452379&fileBranchId=2865970#l27)

File: lib/admin_plugins/futuretreatments.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    var _id = futuretreatments.records[i]._id;
+    $.ajax({
+        method: 'DELETE'
+      , url: '/api/v1/treatments/' + _id
+      , headers: {
+        'api-secret': client.hashauth.hash()
+      }
+    }).done(function success () {
+      $status.text(translate('Record %1 removed ...', { params: [_id] }));
+    }).fail(function fail() {
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futuretreatments.js]: Codacy Issue found: [Don't make functions within a loop.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/983074590/issues/source?bid=2452379&fileBranchId=2865970#l92)

File: lib/admin_plugins/futuretreatments.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+      $status.text(translate('Error removing record %1',  { params: [_id] }));
+    });
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futuretreatments.js]: Codacy Issue found: [Don't make functions within a loop.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/983074590/issues/source?bid=2452379&fileBranchId=2865970#l94)

File: lib/admin_plugins/futureitems.js [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+    var _id = futureitems.treatmentrecords[i]._id;
+    $.ajax({
+        method: 'DELETE'
+      , url: '/api/v1/treatments/' + _id
+      , headers: {
+        'api-secret': client.hashauth.hash()
+      }
+    }).done(function success () {
+      $status.text(translate('Record %1 removed ...', { params: [_id] }));
+    }).fail(function fail() {
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futureitems.js]: Codacy Issue found: [Don't make functions within a loop.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/995071951/issues/source?bid=2452379&fileBranchId=2865970#l97)

File: lib/admin_plugins/futureitems.js [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+      $status.text(translate('Error removing record %1',  { params: [_id] }));
+    });
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futureitems.js]: Codacy Issue found: [Don't make functions within a loop.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/995071951/issues/source?bid=2452379&fileBranchId=2865970#l99)

File: lib/admin_plugins/futureitems.js [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+      , description: 'This task find and remove CGM data in the future created by uploader with wrong date/time.'
+      , buttonLabel: 'Remove entries in the future'
+    }
+  ];
+
+futureitems.actions[0].init = function init(client, callback) {
+  var translate = client.translate;
+  var $status = $('#admin_' + futureitems.name + '_0_status');
+  
+  function showTreatments(treatments, table) {
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futureitems.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/995071951/issues/source?bid=2452379&fileBranchId=2865970#l32)

File: lib/admin_plugins/futureitems.js [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+      .append($('<th>').css('width','80px').attr('align','left').append(translate('Time')))
+      .append($('<th>').css('width','150px').attr('align','left').append(translate('Event Type')))
+      .append($('<th>').css('width','150px').attr('align','left').append(translate('Blood Glucose')))
+      .append($('<th>').css('width','50px').attr('align','left').append(translate('Insulin')))
+      .append($('<th>').css('width','50px').attr('align','left').append(translate('Carbs')))
+      .append($('<th>').css('width','150px').attr('align','left').append(translate('Entered By')))
+      .append($('<th>').css('width','300px').attr('align','left').append(translate('Notes')))
+    );
+    for (var t=0; t<treatments.length; t++) {
+      showOneTreatment (treatments[t], table)
```
Comment (jasoncalabrese) [inline lib/admin_plugins/futureitems.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1008808722/issues/source?bid=2452379&fileBranchId=2865970#l59)

**Comment by jasoncalabrese** [comment]
Works good, we should get a release out, this will help everyone switching over to rest

**Pull request merged**